File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        
        if len(nums) == 2:
            return [0,1]
        
        for i in range(len(nums)):
            if nums[i] in numMap:
                return [numMap[nums[i]],i]
            numMap[target - nums[i]] = i

File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var sumDict = [Int:Int]() // target - num : index
        for (indx,val) in nums.enumerated() {
            if let key = sumDict[val] {
                return [key,indx]
            }
            sumDict[target-val] = indx
        }
        return []
    }
}